zigpy/quirks/registry.py

```diff
@@ -15,27 +15,32 @@ class DeviceRegistry:
     def __init__(self, *args, **kwargs):
         dd = collections.defaultdict
         self._registry = dd(lambda: dd(list))
+        self._model_registry = dd(list)
 
     def add_to_registry(self, custom_device: CustomDeviceType) -> None:
         """Add a device to the registry"""
         models_info = custom_device.signature.get(SIG_MODELS_INFO)
         if models_info:
             for manuf, model in models_info:
                 self.registry[manuf][model].append(custom_device)
+                self._model_registry[model].append(custom_device)
```

Comment:
I see no point duplicating model lists in `self._registry` dict and in `self._models` 
Can you refactor this into a separate PR where `self._registy` is a dict of manufactures with value of an entry in `self._model` dict?

e.g. something like
```
    def __init__(self, *args, **kwargs):
        dd = collections.defaultdict
        self._model_registry = dd(list)
        self._registry = dd(lambda: self._model_registry)
```

then you don't even need to expose the model registry and no need for 
```
        if manufacturer:
            return self.registry[manufacturer][model]
        return self.model_registry[model]
```

a simple `return self._registry[manufacturer][model] would do it`